internal/service/s3/bucket_accelerate_configuration.go

```diff
@@ -62,6 +62,9 @@ func resourceBucketAccelerateConfigurationCreate(ctx context.Context, d *schema.
 	conn := meta.(*conns.AWSClient).S3Client(ctx)
 
 	bucket := d.Get(names.AttrBucket).(string)
+	if isDirectoryBucket(bucket) {
+		conn = meta.(*conns.AWSClient).S3ExpressClient(ctx)
+	}
```

Comment:
Long term I wonder if there is a way to pull this up into `conns` such that a call to `S3Client` could just "do the right thing". Or perhaps leave `S3Client`, but add a custom initializer with a name that better describes that the resulting client will vary depending on the type of bucket (`S3ClientForBucketType(ctx, bucket)`)